File: C++/ITSDA2/Assignment/Question 4/Customer.cpp
#include "Customer.h"

using namespace std;

Customer::Customer()
{
    this->name = "";
    this->phoneNum = "";
    this->email = "";
    this->reservation = nullptr;
}

Customer::Customer(string n, string pn, string em)
{
    this->name = n;
    this->phoneNum = pn;
    this->email = em;
    this->reservation = nullptr;
}

// COMPLETE. WORKS
Customer::~Customer() // DESTRUCTOR
{
    delete reservation;
    reservation = nullptr;

    // cout << "Customer " << this->name << " has been deleted." << endl;
}

Reservation *Customer::getReservation()
{
    return reservation;
}

void Customer::setReservation(Reservation *r)
{
    reservation = new Reservation();
    reservation->setRestaurantName(r->getRestaurantName());
    reservation->setDate(r->getDate());
    reservation->setTableNum(r->getTableNum());
    reservation->setNumPatrons(r->getNumPatrons());
    reservation->setDepositAmount(r->getDepositAmount());
    reservation->setIsDepositPaid(r->getIsDepositPaid());
    reservation->setWaitor(r->getWaitor());
    reservation->setCustomer(this);
}

File: Java/Formative Assessments/Assignment 1/src/Customer.java

public class Customer {

    private String name, email;

    public Customer() {
    }

    public Customer(String n, String e) {
        name = n;
        email = e;
    }

    public void setName(String n) {
        name = n;
    }

    public String getName() {
        return name;
    }

    public void setEmail(String e) {
        email = e;
    }

    public String getEmail() {
        return email;
    }
}
